pkg/execution/executor/executor.go

```diff
@@ -306,6 +314,12 @@ func (e *executor) run(ctx context.Context, w inngest.Workflow, id state.Identif
 		return nil, idx, err
 	}
 	if response.Err != nil {
+		if response.Final() && e.failureHandler != nil {
+			if err := (*e.failureHandler)(ctx, id, s, *response); err != nil {
```

Comment:
When this isn't a pointer you can literally just `if err := e.failureHandler(ctx, id, s, *response)` :)